python/tvm/script/builder/tir/for_frame.py

```diff
@@ -29,26 +29,47 @@
 class ForFrame(TIRFrame):
     def __enter__(self) -> List[Var]:
         _base_ffi_api.FrameEnter(self)  # pylint: disable=no-member # type: ignore
-        return self.vars
+        return self.vars if len(self.vars) > 1 else self.vars[0]
 
 
-def serial(start, stop, annotations=None) -> ForFrame:
+def serial(start, stop=None, annotations=None) -> ForFrame:
```

Comment:
Let's use "*" to enforce users to explicitly write `annotations=` when specifying annotations: https://stackoverflow.com/questions/53797057/star-as-an-argument-in-python-function

```suggestion
def serial(start, stop=None, *, annotations=None) -> ForFrame:
```